usl_models/usl_models/flood_ml/model.py

```diff
@@ -39,6 +34,7 @@ def __init__(
         self,
         model_params: FloodModelParams,
         spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
+        artifact_uri: str | None = None,
```

Comment:
Might suggest something like
```python
def __init__(self, model=None):
  self._model = model or self._build_model()

@classmethod
def from_checkpoint(cls, artifact_uri):
  model = keras.models.load_model(artifact_uri)
  return cls(model=model)
```

Reply:
How would this API support changing the parameters?

Reply:
you can add model parameters to the above methods -- the crux of what I'm suggesting is having an explicit
```
@classmethod
def from_checkpoint
```
method for loading a model from GCS rather than overloading the constructor with additional behaviours

Reply:
Done.